l10n_de_tax_statement/tests/test_l10n_de_tax_statement.py

```diff
@@ -0,0 +1,354 @@
+
+    def test_09_update_working(self):
+        self.invoice_1._onchange_invoice_line_ids()
+        self.invoice_1.action_invoice_open()
+        self.statement_1.statement_update()
+        self.assertEqual(len(self.statement_1.line_ids.ids), 47)
+
+        _25 = self.statement_1.line_ids.filtered(lambda r: r.code == '25')
+        _26 = self.statement_1.line_ids.filtered(lambda r: r.code == '26')
+        _27 = self.statement_1.line_ids.filtered(lambda r: r.code == '27')
+
+        self.assertEqual(len(_25), 1)
+        self.assertEqual(len(_26), 1)
+        self.assertEqual(len(_27), 1)
+
+        self.assertFalse(_25.format_base)
+        self.assertFalse(_25.format_tax)
+        self.assertTrue(_25.is_group)
+        self.assertTrue(_25.is_readonly)
+
+        self.assertEqual(_26.format_base, '100.00')
+        self.assertFalse(_26.is_group)
+        self.assertTrue(_26.is_readonly)
+
+        self.assertEqual(_27.format_base, '50.00')
+        self.assertFalse(_27.is_group)
+        self.assertTrue(_27.is_readonly)
+
+        self.assertEqual(self.statement_1.tax_total, 22.5)
+        self.assertEqual(self.statement_1.format_tax_total, '22.50')
+
+    def test_10_line_unlink_exception(self):
+        self.assertEqual(len(self.statement_1.line_ids.ids), 0)
+        self.assertEqual(self.statement_1.tax_total, 0.)
+
+        self.invoice_1.action_invoice_open()
+        self.statement_1.statement_update()
+        self.statement_1.post()
+        with self.assertRaises(UserError):
+            self.statement_1.line_ids.unlink()
+
+        self.assertEqual(len(self.statement_1.line_ids.ids), 47)
+        self.assertEqual(self.statement_1.tax_total, 22.5)
+
+        for line in self.statement_1.line_ids:
+            self.assertTrue(line.view_base_lines())
+            self.assertTrue(line.view_tax_lines())
+            self.assertTrue(line.is_readonly)
+            with self.assertRaises(UserError):
+                line.unlink()
+
+    def test_11_wizard_execute(self):
+        wizard = self.Wizard.create({})
+
+        self.assertEqual(wizard.tag_81_base, self.tag_1)
+        self.assertEqual(wizard.tag_86_base, self.tag_2)
+
+        wizard.write({
+            'tag_81_base': self.tag_1.id,
+            'tag_86_base': self.tag_2.id,
+        })
+
+        self.config.unlink()
+
+        wizard_2 = self.Wizard.create({})
+        self.assertNotEqual(wizard_2.tag_81_base, self.tag_1)
+        self.assertNotEqual(wizard_2.tag_86_base, self.tag_2)
+
+        config = self.env['l10n.de.tax.statement.config'].search(
+            [('company_id', '=', self.env.user.company_id.id)],
+            limit=1)
+        self.assertFalse(config)
+
+        wizard.execute()
+
+        config = self.env['l10n.de.tax.statement.config'].search(
+            [('company_id', '=', self.env.user.company_id.id)],
+            limit=1)
+        self.assertTrue(config)
+        self.assertEqual(config.tag_86_base, self.tag_1)
```

Comment:
```suggestion
        self.assertEqual(config.tag_81_base, self.tag_1)
```

Reply:
I have applied your proposed change. Thanks a lot for your intervention!